src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeEditorService.java

```diff
@@ -0,0 +1,100 @@
+import gov.cabinetoffice.gap.adminbackend.entities.GrantAdmin;
+import gov.cabinetoffice.gap.adminbackend.entities.SchemeEntity;
+import gov.cabinetoffice.gap.adminbackend.enums.SchemeEditorRoleEnum;
+import gov.cabinetoffice.gap.adminbackend.repositories.SchemeRepository;
+import gov.cabinetoffice.gap.adminbackend.services.encryption.AwsEncryptionServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class SchemeEditorService {
+    private final RestTemplate restTemplate;
+    private final SchemeService schemeService;
+    private final AwsEncryptionServiceImpl awsEncryptionService;
+    private final SchemeRepository schemeRepo;
+    private final UserServiceConfig userServiceConfig;
+
+    @Value("${user-service.domain}")
+    private String userServiceUrl;
+
+    public Boolean doesAdminOwnScheme(Integer schemeId, Integer adminId) {
+        SchemeEntity scheme = schemeService.findSchemeById(schemeId);
+        List<SchemeEntity> adminSchemes = schemeRepo.findByCreatedBy(adminId);
+        if (adminSchemes.contains(scheme)) {
+            return Boolean.TRUE;
+        }
+        return Boolean.FALSE;
+    }
+
+    public List<SchemeEditorsDTO> getEditorsFromSchemeId(Integer schemeId, String authHeader) {
+        SchemeEntity scheme = schemeService.findSchemeById(schemeId);
+        List<GrantAdmin> editors = scheme.getGrantAdmins();
+        Integer createdBy = scheme.getCreatedBy();
+
+        return this.mapEditorListToDto(editors, createdBy, authHeader);
+    }
+
+    private List<SchemeEditorsDTO> mapEditorListToDto(List<GrantAdmin> editors, Integer createdBy, String authHeader){
+        List<String> userSubs = editors.stream()
+                .map(editor -> editor.getGapUser().getUserSub())
+                .toList();
+        List<String> emails = getEmailsFromUserSubBatch(userSubs, authHeader);
+
+        return editors.stream()
+                .map(editor -> {
+                    String email = emails.get(userSubs.indexOf(editor.getGapUser().getUserSub()));
+                    Integer id = editor.getId();
+                    SchemeEditorRoleEnum role = id.equals(createdBy)
+                            ? SchemeEditorRoleEnum.Owner : SchemeEditorRoleEnum.Editor;
+
+                    return SchemeEditorsDTO.builder().id(id).email(email).role(role).build();
+                })
+                .sorted(Comparator.comparing(SchemeEditorsDTO::role, Comparator.reverseOrder()))
+                .toList();
```

Comment:
A small refactor for this might be to make `getEmailsFromUserSubBatch` also return the users sub so we don't have to do that horrible matching up of emails to userSubs

Reply:
it's a bit complex, but since we already have the user-subs at this point i don't see the need to refetch them just for readability?

Reply:
yeah fair - maybe refactor that logic into a well named function then. Something like:
```
    private String marryEmailFromSub(final List<String> emails, final List<String> userSubs, final GrantAdmin editor) {
        return emails.get(userSubs.indexOf(editor.getGapUser().getUserSub()));
    }